src/app/assessment/assessment.component.ts

```diff
@@ -205,17 +205,34 @@ export class AssessmentComponent extends RouterEnter {
     return this.router.navigate(['app', 'home']);
   }
 
-  // form an object of required questions
-  getRequiredQuestions() {
+  /**
+   * @name compulsoryQuestionsAnswered
+   * @description to check if every compulsory question has been answered
+   * @param {Object[]} answers a list of answer object (in submission-based format)
+   */
+  compulsoryQuestionsAnswered(answers) {
+    const result = [];
     const requiredQuestions = {};
+    const missing = [];
     this.assessment.groups.forEach(group => {
       group.questions.forEach(question => {
         if (question.isRequired) {
           requiredQuestions[question.id] = true;
         }
       });
     });
-    return requiredQuestions;
+
+    this.utils.each(requiredQuestions, (question, questionId) => {
+      if (missing.length === 0) {
```

Comment:
What I mean from last comment is to put this logic inside the above loop, so that we don't need to loop `requiredQuestions` again. We don't actually need `requiredQuestions` this variable
```
this.assessment.groups.forEach(group => {
  group.questions.forEach(question => {
    if (question.isRequired) {
      if (missing.length === 0) {
   .....
    }
  });
});
```